src/_articles/archive/zones.md

```diff
@@ -152,33 +154,51 @@ Code might execute in other nested child zones as well,
 but at a minimum it always runs in the root zone.
 
 
-## Handling asynchronous errors
+## Handling uncaught errors
 
 One of the most used features of zones is
-their ability to handle uncaught errors
-in asynchronously executing code.
-The concept is similar to a try-catch in synchronous code.
-
-An _uncaught error_ is often caused by some code using `throw`
-to raise an exception that is not handled by a `catch` statement.
-Another way to produce an uncaught error is
-to call `new Future.error()` or
-a Completer's `completeError()` method.
-
-Use the `onError` argument to `runZoned()` to
-install a _zoned error handler_—an asynchronous error handler
-that's invoked for every uncaught error in the zone.
-For example:
+their ability to catch and handle uncaught errors,
+both synchronous and asynchronous.
+An _uncaught error_ is often caused by code using `throw`
+to raise an exception without an accompanying `catch`
+statement to handle it.
+
+In `async` functions, errors usually end up as the result of a Future.
+Futures allow you to handle errors where you `await` the future.
+Sometimes an error escapes, or a future is not `await`-ed,
+in which case an error becomes an uncaught error.
+Uncaught errors can crash your program.
+
+Zones can install an _uncaught error handler_ upon creation.
+When an uncaught error occurs in that zone,
+rather than just crashing the program,
+it calls its uncaught error handler.
+(In fact, any zone that doesn't specify a handler
+inherits the uncaught error handler of the root zone,
+which just crashes the program on an uncaught error).
```

Comment:
> That last sentence didn't make sense. If the zone inherits the root zone's UEH, then why would it crash at all? Does it crash if the root zone doesn't have an UEH?

This was a little confusing for me to wrap my head around too, and they way I got it out definitely isn't ideal. 

Root zones have an implicit (or maybe "default" or "built-in" is a better term) uncaught error handler. That UEH also handles uncaught errors: it handles them by crashing the program. So it's saying that user-created zones inherit that implicit UEH from the root, and to explicitly define one is to override the root's implicit one, to do something other than crash (is that correct @lrhn ?)

To me, it's like a neat little implementation detail of the language (which is why it's in parenthesis, it actually can probably be ignored). Like saying that even crashing is a form of error handling because Dart had to be designed to react, that reaction doesn't come out of no where.

Reply:
Looking again, idk why I rewrote it the way I did because Lasse's original wording might make it more clear, what do you think?

> (In fact, the code which crashes the program on an uncaught error
is just the uncaught error handler of the root zone,
which any zone inherits if it doesn’t specify another handler).

Reply:
**reply:** This makes sense, but reads like a meandering path. Maybe:
```markdown
The root zone handles any uncaught errors when another zone
lacks an explicit uncaught error handler.
```

I will cop to the fact that I despise parentheticals in technical documentation. Asides make little sense in these contexts. Explain what needs to be understood. If it's worth writing, it's worth writing as if you intend it to be read.